hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/ConfigurationHotUpdateStrategy.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.utilities.deltastreamer;
+
+import org.apache.hudi.common.config.TypedProperties;
+
+/**
+ * Configuration of delta sync hot update strategy.
+ */
+public interface ConfigurationHotUpdateStrategy {
+  /**
+   * Config hot update. The method is called in main on driver, so we can update the properties in place.
+   * @param properties props may be updated.
```

Comment:
There is no ensurance that the class always have constructor params like
```java
SparkSession sparkSession,
HoodieDeltaStreamer.Config cfg,
TypedProperties properties
```

Reply:
So do you mean I should make the interface an abstract class? like 
```java
public abstract class ConfigurationHotUpdateStrategy {

  protected HoodieDeltaStreamer.Config cfg;
  protected TypedProperties properties;
  
  public ConfigurationHotUpdateStrategy(HoodieDeltaStreamer.Config cfg,
                                        TypedProperties properties) {
    this.cfg = cfg;
    this.properties = properties;
  }

  /**
   * Config hot update. The method is called in main on driver, so we can update the properties in place.
   * @param properties props may be updated.
   * @return true if props updated.
   */
  boolean updateProps(final TypedProperties properties) {
    return false;
  }
}
```

Reply:
Yeah, probably.